### PR TITLE
feat: implement cpu_time subroutine and removed from old runtime lib

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Continuing: https://github.com/lfortran/lfortran/pull/4866

### CODE REVIEW

**Comment by HarshitaKalani** [comment]
I'm not sure of the wasm failure here. @Shaikh-Ubaid can you have a look and guide please?

File: tests/reference/wat-cpu_time_02_wasm-fa2b15f.stderr [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    { return std::forward<_Fn>(__f)(std::forward<_Args>(__args)...); }
+[2m  File "[0m[1m[35m$DIR/src/libasr/codegen/asr_to_wasm.cpp[39m[0m[2m", line 1143, in LCompilers::ASRToWASMVisitor::emit_function_body(LCompilers::ASR::Function_t const&)::'lambda'()::operator()() const[0m
+    this->visit_stmt(*x.m_body[i]);
+[2m  File "[0m[1m[35m$DIR/src/libasr/../libasr/asr.h[39m[0m[2m", line 5179, in LCompilers::ASR::BaseVisitor<LCompilers::ASRToWASMVisitor>::visit_stmt(LCompilers::ASR::stmt_t const&)[0m
+    void visit_stmt(const stmt_t &b) { visit_stmt_t(b, self()); }
+[2m  File "[0m[1m[35m$DIR/src/libasr/../libasr/asr.h[39m[0m[2m", line 4927, in void LCompilers::ASR::visit_stmt_t<LCompilers::ASRToWASMVisitor>(LCompilers::ASR::stmt_t const&, LCompilers::ASRToWASMVisitor&)[0m
+    case stmtType::IntrinsicImpureSubroutine: { v.visit_IntrinsicImpureSubroutine((const IntrinsicImpureSubroutine_t &)x); return; }
+LCompilersException: visit_IntrinsicImpureSubroutine() not implemented
```
Comment (Shaikh-Ubaid) [inline tests/reference/wat-cpu_time_02_wasm-fa2b15f.stderr:56]: Here is the issue. The generated ASR now includes `visit_IntrinsicImpureSubroutine()`, which is not yet implemented in the wasm backend.

I think the possible solutions could be:
- Implement the `visit_IntrinsicImpureSubroutine()` in the wasm backend
- Or for now to unblock you, I think we can disable the `cpu_time` tests for wasm. I think there would be 2-3 such tests (including integration and reference tests). (We don't usually follow this, but I think it should be fine for now).

We can later look into implementing the `visit_IntrinsicImpureSubroutine` for wasm.

Comment (HarshitaKalani) [reply tests/reference/wat-cpu_time_02_wasm-fa2b15f.stderr:56]: I see. Thanks a lot!

**Comment by certik** [comment]
If WASM is the only problem then just comment out the failing tests for it. @Shaikh-Ubaid can fix it later.

**Comment by certik** [comment]
Last error in the generated C++ code:
```
CMakeFiles/matmul_01.dir/matmul_01.f90.o.tmp.c:70:8: error: conflicting types for ‘_lfortran_d_cpu_time’; have ‘double(double *)’
   70 | double _lfortran_d_cpu_time(double *n);
      |        ^~~~~~~~~~~~~~~~~~~~
In file included from CMakeFiles/matmul_01.dir/matmul_01.f90.o.tmp.c:8:
```

File: src/libasr/runtime/lfortran_intrinsics.c [hunk 2]
```diff
@@ -2409,14 +2409,14 @@ LFORTRAN_API int64_t _lfortran_ibits64(int64_t i, int32_t pos, int32_t len) {
...
+    return *t;
 }
 
-LFORTRAN_API float _lfortran_s_cpu_time(float t) {
-    t = ((float) clock()) / CLOCKS_PER_SEC;
-    return t;
+LFORTRAN_API float _lfortran_s_cpu_time(float *t) {
+    *t = ((float) clock()) / CLOCKS_PER_SEC;
+    return *t;
 }
```
Comment (certik) [inline src/libasr/runtime/lfortran_intrinsics.c:2420]: This can be fixed later, but why do these functions accept `t` and return `t`?

Comment (HarshitaKalani) [reply src/libasr/runtime/lfortran_intrinsics.c:2418]: Oh yes, I didn't check this. Thanks for bringing this into my notice. Corrected this now.

**Review by certik — APPROVED** [review]
Yes, there is a regression in WASM, however it removes technical debt and we can fix WASM later. Maybe @Shaikh-Ubaid can fix it. :)

Thanks for the PR, I think this is good to go in.

**Pull request merged**